frontend/src/redux/clusterActionSlice.ts

```diff
@@ -242,7 +242,7 @@ export const executeClusterAction = createAsyncThunk(
         if (controller.signal.aborted) {
           return rejectWithValue('Action cancelled');
         }
-        callback();
+        await callback();
```

Comment:
Will this not break if callbacks are not promises? We should double check this.
If this logic is internal so far, we can alter the cases where it's being called and convert those functions to async ones, but we need to double check nonetheless.

Reply:
I believe just wrapping `callback()` with a Promise should address this, they seem to be functionally equivalent in how they send failure/success messages